src/python/WMCore/Services/DDM/DDM.py

```diff
@@ -0,0 +1,309 @@
+from __future__ import (division, print_function)
+
+import json
+import logging
+
+from WMCore.Services.Service import Service
+from pprint import pformat
+
+
+class DDMReqTemplate(dict):
+    """
+    A simple class defining a set of request templates for DDM, with default
+    structures representing the different POST calls expected by the modules
+    here:
+        https://github.com/SmartDataProjects/dynamo/tree/master/lib/web/modules
+    """
+
+    def __init__(self, api=None, **kwargs):
+        """
+        api:
+            String representing the type of request to be sent. In case the api
+            is missing or unrecognized the created object will be an ordinary
+            dictionary.
+        example:
+            ddmReq = DDMReqTemplate('copy',
+                                     item=['/LQLQToTop..._M-1200/NANOAODSIM'],
+                                     site=['T2_CH_CERN', 'T2_US_MIT'])
+        NOTE: we do not enforce the use of api parameter, but we may change this
+        """
+        super(DDMReqTemplate, self).__init__(**kwargs)
+
+        self.api = api
+        if api is None:
+            msg = "ERROR: Missing API parameter"
+            raise ValueError(msg)
+
+        # tempTuple format: [(key, defaultValue, type), ....()]
+        if api == 'copy':
+            tempTuple = [('item', [], list),
+                         ('site', ['T2_*', 'T1_*_Disk'], list),
+                         ('group', 'DataOps', str),
+                         ('n', None, int),
+                         ('cache', None, str)]
+
+        elif api == 'pollcopy':
+            tempTuple = [('request_id', None, int),
+                         ('item', None, list),
+                         ('site', None, list),
+                         ('status', None, str),
+                         ('user', None, str)]
+
+        elif api == 'cancelcopy':
+            tempTuple = [('request_id', None, int)]
+
+        else:
+            msg = "ERROR: Unsupported API: {}".format(api)
+            raise ValueError(msg)
+
+        # buildTemplate:
```

Comment:
I was thinking about that before and I was playing around with stripping the `None` values as early as possible, but this can break things in two ways. First the needed properties to the request may be added on a later stage, And I would like to have it done not as an update to the dictionary but just with an overwrite of an existing placeholder (we may need to enforce type checking if a request parameter is added later). The second is it can easily break the isEqual method... because one may try to aggregate on a key which was striped earlier, so it will exclude  that key from the check and will generate a false positive. That's the reason why (as mentioned in my comment to the previous conversation), we must keep the static structure of this template (with the `None` values in it) until the ferry last moment just before we send it to DDM. Ergo we cannot simplify the code here.